src/components/SignIn.js

```diff
@@ -0,0 +1,61 @@
+import React, { Fragment, useRef, useState } from "react"
+import { Form, Button, Card, Alert } from "react-bootstrap"
+import { useAuth } from "../contexts/AuthContext"
+import { Link, useHistory } from "react-router-dom"
+import * as ROUTES from './routes';
+
+ 
+const SignIn = () => {
+  const emailRef = useRef()
+  const passwordRef = useRef()
+  const {login} = useAuth()
+  const [error, setError] = useState("")
+  const [loading, setLoading] = useState(false)
+  const history = useHistory()
+
+  async function handleSubmit(e) {    
+    e.preventDefault()
+    try {
+      setError("")
+      setLoading(true)
+       await login(emailRef.current.value, passwordRef.current.value)
+       history.push("/createBlog")
+    } catch {
+      setError("Failed to login")
+    }
+    setLoading(false)
```

Comment:
This is probably one of the times that's causing your messages about updating the state on an unmounted component. The `history.push` call will redirect you and cause this component to unmount. After that's happened, the `setLoading` call will update the state on the now unmounted component. 

To get around this, I would consider just calling `setLoading(false)` in the `catch` since when you re-mount this component it will reset the state to the default value, `false`.